crates/ruff/src/rules/pylint/rules/compare_to_empty_string.rs

```diff
@@ -0,0 +1,62 @@
+use itertools::Itertools;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::{unparse_constant, unparse_expr};
+use rustpython_parser::ast::{Cmpop, Constant, Expr, ExprKind, Located};
+
+use crate::{checkers::ast::Checker, registry::Diagnostic, violation::Violation};
+
+use super::comparison_of_constant::ViolationsCmpop;
+
+#[violation]
+pub struct CompareToEmptyString {
+    pub lhs: String,
+    pub op: ViolationsCmpop,
+    pub rhs: String,
+}
+
+impl Violation for CompareToEmptyString {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let cond = match self.op {
+            ViolationsCmpop::Is | ViolationsCmpop::Eq => "",
+            // the assumption is that this message will only ever be shown
+            // if op is `in`, `=`, `not in`, `!=`
+            _ => "not",
```

Comment:
I'd suggest just making a new `enum` in this file specifically for this violation, to encode this assumption in the types themselves.